### PR TITLE
Add ability to build docker image from Dockerfile

### PR DESCRIPTION
**Pull request by dhodovsk**
Fixes: https://github.com/fedora-modularity/conu/issues/77

If you think another option from https://github.com/docker/docker-py/blob/master/docker/api/build.py#L60 should be added, please, let me know.

### CODE REVIEW

File: tests/integration/test_docker.py [hunk 1]
```diff
@@ -374,3 +374,12 @@ def test_list_images():
         assert len(image_list) > 0
         assert image_list[0].short_metadata
         assert image_list[0].short_metadata["Id"]
+
+
+def test_build_image():
+    with DockerBackend() as backend:
+        name = 'conu-test-image'
+        image = backend.ImageClass.build('/home/dhodovsk/repos/github.com/fedora-modularity/conu/tests/integration/data/greeter', tag=name)
```
Comment (lachmanfrantisek) [inline tests/integration/test_docker.py]: It would be nice if we can have some example dockerfile in the repo..;-)

Comment (dhodovsk) [reply tests/integration/test_docker.py]: Heh, sorry this is just a typo, it should be of course `os.getcwd()` or similar, thanks for the catch :)

File: conu/backend/docker/image.py [hunk 2]
```diff
@@ -418,6 +418,30 @@ def get_volume_options(volumes):
...
+        :return: instance of DockerImage
+        """
+        if not path:
+            raise ConuException('Please specify path to the directory containing the Dockerfile')
+        client = get_client()
+        response = [line for line in client.build(path,
+                                                  rm=True, tag=tag,
+                                                  dockerfile=dockerfile,
+                                                  quiet=True)]
+        if not len(response):
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: `if not response:` should be enough

File: conu/backend/docker/image.py [hunk 3]
```diff
@@ -418,6 +418,30 @@ def get_volume_options(volumes):
...
+            raise ConuException('Failed to get ID of image')
+
+        # get ID from output
+        # b'{"stream":"sha256:39c7bac4e2da37983203df4fcf612a02de9e6f6456a7f3434d1fccbc9ad639a5\\n"}\r\n'
+        image_id = response[0].decode('utf-8')[11:-6]
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: this looks pretty fragile, no idea whether we should add some error handling in it

**Comment by TomasTomecek** [comment]
CI, where are you?

**Comment by lachmanfrantisek** [comment]
@TomasTomecek  I'm ok with that. The [build method in new docker-py](https://github.com/docker/docker-py/blob/9ff787cb5f6b2ad21669c54a7fa089498538dd2a/docker/models/images.py#L186) does not handle that much differently.

**Pull request merged**